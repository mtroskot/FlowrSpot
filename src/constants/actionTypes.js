export const userActionTypes = Object.freeze({
  REGISTER_USER: 'REGISTER_USER',
  LOGIN_USER: 'LOGIN_USER',
  LOGIN_USER_SUCCESS: 'LOGIN_USER_SUCCESS',
  LOGOUT_USER: 'LOGOUT_USER'
});

export const flowerActionTypes = Object.freeze({
  GET_FAVORITE_FLOWERS: 'GET_FAVORITE_FLOWERS',
  GET_FAVORITE_FLOWERS_SUCCESS: 'GET_FAVORITE_FLOWERS_SUCCESS',
  FAVORITE_FLOWER: 'FAVORITE_FLOWER',
  FAVORITE_FLOWER_SUCCESS: 'FAVORITE_FLOWER_SUCCESS',
  UNFAVORITE_FLOWER: 'UNFAVORITE_FLOWER',
  UNFAVORITE_FLOWER_SUCCESS: 'UNFAVORITE_FLOWER_SUCCESS'
});

export const uiActionTypes = Object.freeze({
  START_ACTION: 'START_ACTION',
  STOP_ACTION: 'STOP_ACTION',
  REFRESH_ACTION_START: 'REFRESH_ACTION_START',
  REFRESH_ACTION_STOP: 'REFRESH_ACTION_STOP',
  TOGGLE_POPUP_MESSAGE: 'TOGGLE_POPUP_MESSAGE'
});
