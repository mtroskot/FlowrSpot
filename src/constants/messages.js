export const DEFAULT_ERROR = 'Something went wrong';
export const NO_RESULTS = 'No Results Found';
export const REGISTRATION_SUCCESS = 'Registration Success';
