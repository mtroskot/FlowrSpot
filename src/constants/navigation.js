export const screenNames = Object.freeze({
  COMMENTS: 'COMMENTS',
  FLOWER_DETAILS: 'FLOWER_DETAILS',
  FLOWERS: 'FLOWERS',
  LATEST_SIGHTINGS: 'LATEST_SIGHTINGS',
  FAVORITES: 'FAVORITES',
  SETTINGS: 'SETTINGS',
  LOGIN: 'LOGIN',
  REGISTER: 'REGISTER',
  HOME: 'HOME',
  LOCATION: 'LOCATION',
  MAP: 'MAP',
  MENU: 'MENU'
});

export const stackNames = Object.freeze({
  COMMENTS_STACK: 'COMMENTS_STACK',
  HOME_STACK: 'HOME_STACK',
  LOCATION_STACK: 'LOCATION_STACK',
  MAP_STACK: 'MAP_STACK',
  MENU_STACK: 'MENU_STACK'
});
