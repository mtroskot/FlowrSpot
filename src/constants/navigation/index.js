export const screenNames = Object.freeze({
  HOME: 'HOME'
});
