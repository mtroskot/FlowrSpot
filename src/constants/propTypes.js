import PropTypes from 'prop-types';

export const flowerPropTypes = PropTypes.exact({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  latin_name: PropTypes.string.isRequired,
  sightings: PropTypes.number.isRequired,
  profile_picture: PropTypes.string.isRequired,
  favorite: PropTypes.bool.isRequired
});

export const userPropTypes = PropTypes.exact({
  id: PropTypes.number,
  email: PropTypes.string,
  authToken: PropTypes.string,
  firstName: PropTypes.number,
  lastName: PropTypes.string
});

export const popupMessagePropTypes = PropTypes.exact({
  message: PropTypes.string,
  position: PropTypes.oneOf(['bottom', 'top'])
});

export const favoriteFlowerListPropTypes = PropTypes.arrayOf(
  PropTypes.exact({
    flowerId: PropTypes.number.isRequired,
    favoriteId: PropTypes.number.isRequired
  })
);
