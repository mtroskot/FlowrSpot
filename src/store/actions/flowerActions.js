import { flowerActionTypes } from 'src/constants/actionTypes';

export const getFavoriteFlowers = () => ({
  type: flowerActionTypes.GET_FAVORITE_FLOWERS,
  payload: {}
});

export const getFavoriteFlowersSuccess = favoriteFlowers => ({
  type: flowerActionTypes.GET_FAVORITE_FLOWERS_SUCCESS,
  payload: { favoriteFlowers }
});

export const favoriteFlower = flowerId => ({
  type: flowerActionTypes.FAVORITE_FLOWER,
  payload: { flowerId }
});

export const favoriteFlowerSuccess = (flowerId, favoriteId) => ({
  type: flowerActionTypes.FAVORITE_FLOWER_SUCCESS,
  payload: { flowerId, favoriteId }
});

export const unfavoriteFlower = (flowerId, favoriteId) => ({
  type: flowerActionTypes.UNFAVORITE_FLOWER,
  payload: { flowerId, favoriteId }
});

export const unfavoriteFlowerSuccess = flowerId => ({
  type: flowerActionTypes.UNFAVORITE_FLOWER_SUCCESS,
  payload: { flowerId }
});
