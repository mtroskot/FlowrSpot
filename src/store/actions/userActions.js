import { userActionTypes } from 'src/constants/actionTypes';

export const registerUser = registerData => ({
  type: userActionTypes.REGISTER_USER,
  payload: { registerData }
});

export const loginUser = (email, password) => ({
  type: userActionTypes.LOGIN_USER,
  payload: { email, password }
});

export const loginUserSuccess = (id, email, authToken, firstName, lastName) => ({
  type: userActionTypes.LOGIN_USER_SUCCESS,
  payload: { id, email, authToken, firstName, lastName }
});

export const logoutUser = () => ({
  type: userActionTypes.LOGOUT_USER,
  payload: {}
});
