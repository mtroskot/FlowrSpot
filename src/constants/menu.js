import { screenNames } from 'src/constants/navigation';

export const menuList = Object.freeze([
  {
    menuId: screenNames.FLOWERS,
    text: 'Flowers'
  },
  {
    menuId: screenNames.LATEST_SIGHTINGS,
    text: 'Latest Sightings'
  },
  {
    menuId: screenNames.FAVORITES,
    text: 'Favorites'
  },
  {
    menuId: screenNames.SETTINGS,
    text: 'Settings'
  },
  {
    menuId: screenNames.LOGIN,
    text: 'Login'
  }
]);
