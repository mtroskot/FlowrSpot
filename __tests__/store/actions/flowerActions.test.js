import configureStore from 'redux-mock-store';
import * as flowerActions from 'src/store/actions/flowerActions';
import { flowerActionTypes } from 'src/constants/actionTypes';

const middlewares = [];
const mockStore = configureStore(middlewares);
// Initialize mockstore with empty state
const initialState = {};
const store = mockStore(initialState);

describe('Flower actions tests', () => {
  beforeEach(() => {
    store.clearActions();
  });

  it('dispatches getFavoriteFlowers action', () => {
    const expectedActions = [
      {
        type: flowerActionTypes.GET_FAVORITE_FLOWERS,
        payload: {}
      }
    ];
    store.dispatch(flowerActions.getFavoriteFlowers());
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('dispatches getFavoriteFlowersSuccess action', () => {
    const favoriteFlowers = [];
    const expectedActions = [
      {
        type: flowerActionTypes.GET_FAVORITE_FLOWERS_SUCCESS,
        payload: { favoriteFlowers }
      }
    ];
    store.dispatch(flowerActions.getFavoriteFlowersSuccess(favoriteFlowers));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
  it('dispatches favoriteFlower action', () => {
    const flowerId = 'flowerId';
    const expectedActions = [
      {
        type: flowerActionTypes.FAVORITE_FLOWER,
        payload: { flowerId }
      }
    ];
    store.dispatch(flowerActions.favoriteFlower(flowerId));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
  it('dispatches favoriteFlowerSuccess action', () => {
    const flowerId = 'flowerId';
    const favoriteId = 'favoriteId';
    const expectedActions = [
      {
        type: flowerActionTypes.FAVORITE_FLOWER_SUCCESS,
        payload: { flowerId, favoriteId }
      }
    ];
    store.dispatch(flowerActions.favoriteFlowerSuccess(flowerId, favoriteId));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
  it('dispatches unfavoriteFlower action', () => {
    const flowerId = 'flowerId';
    const favoriteId = 'favoriteId';
    const expectedActions = [
      {
        type: flowerActionTypes.UNFAVORITE_FLOWER,
        payload: { flowerId, favoriteId }
      }
    ];
    store.dispatch(flowerActions.unfavoriteFlower(flowerId, favoriteId));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
  it('dispatches unfavoriteFlowerSuccess action', () => {
    const flowerId = 'flowerId';
    const expectedActions = [
      {
        type: flowerActionTypes.UNFAVORITE_FLOWER_SUCCESS,
        payload: { flowerId }
      }
    ];
    store.dispatch(flowerActions.unfavoriteFlowerSuccess(flowerId));
    expect(store.getActions()).toEqual(expectedActions);
    expect(store.getActions()).toMatchSnapshot();
  });
});
