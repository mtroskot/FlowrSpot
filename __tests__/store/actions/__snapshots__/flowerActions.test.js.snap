// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flower actions tests dispatches favoriteFlower action 1`] = `
Array [
  Object {
    "payload": Object {
      "flowerId": "flowerId",
    },
    "type": "FAVORITE_FLOWER",
  },
]
`;

exports[`Flower actions tests dispatches favoriteFlowerSuccess action 1`] = `
Array [
  Object {
    "payload": Object {
      "favoriteId": "favoriteId",
      "flowerId": "flowerId",
    },
    "type": "FAVORITE_FLOWER_SUCCESS",
  },
]
`;

exports[`Flower actions tests dispatches getFavoriteFlowers action 1`] = `
Array [
  Object {
    "payload": Object {},
    "type": "GET_FAVORITE_FLOWERS",
  },
]
`;

exports[`Flower actions tests dispatches getFavoriteFlowersSuccess action 1`] = `
Array [
  Object {
    "payload": Object {
      "favoriteFlowers": Array [],
    },
    "type": "GET_FAVORITE_FLOWERS_SUCCESS",
  },
]
`;

exports[`Flower actions tests dispatches unfavoriteFlower action 1`] = `
Array [
  Object {
    "payload": Object {
      "favoriteId": "favoriteId",
      "flowerId": "flowerId",
    },
    "type": "UNFAVORITE_FLOWER",
  },
]
`;

exports[`Flower actions tests dispatches unfavoriteFlowerSuccess action 1`] = `
Array [
  Object {
    "payload": Object {
      "flowerId": "flowerId",
    },
    "type": "UNFAVORITE_FLOWER_SUCCESS",
  },
]
`;
